### PR TITLE
created CustomGlobalExceptionHandler

### PR DESCRIPTION
**Pull request by StoneBlood-bit**

### CODE REVIEW

File: src/main/java/mate/academy/controller/BookController.java [hunk 1]
```diff
@@ -29,8 +29,8 @@ public BookDto getBookById(@PathVariable Long id) {
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
-        return bookService.save(bookDto);
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
+        return bookService.save(requestDto);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/controller/BookController.java:33]: Don't provide BookDto in updateBook endpoint, create new dto or use CreateBookRequestDto
Also, add validation for this dto

File: src/main/java/mate/academy/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,13 +1,20 @@
 package mate.academy.dto;
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/dto/CreateBookRequestDto.java]: for String use `@NotBlank`

```suggestion
    @NotBlank
```

File: src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        return new ResponseEntity<>(body, headers, HttpStatus.BAD_REQUEST);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by sarakhmen — APPROVED** [review]

**Pull request merged**